swan-lake/development-tutorials/test-document-the-code/test-ballerina-code/mocking.md

```diff
@@ -211,6 +207,191 @@ public function testGetRandomJoke() {
+import ballerina/email;
+
+@test:Config {}
+function testSendNotification() {
+    string[] emailIds = ["user1@test.com", "user2@test.com"];
+
+    // Create a default mock SMTP client and assign it to the `smtpClient` object.
+    smtpClient = test:mock(email:SmtpClient);
+
+    // Stub to do nothing when the`sendMessage` method is invoked.
+    test:prepare(smtpClient).when("sendMessage").doNothing();
+
+    // Invoke the function to test and verify that no error occurred.
+    test:assertEquals(sendNotification(emailIds), ());
+}
+```
+
+#### Stub resource methods
+
+Similar to remote methods, resource methods also can be stubbed to return a specific value, a series of values or, to do nothing. To mock a resource method, the resource method name(`get`, `post`, `put`, ...etc) and resource path should be specified. Each path parameter in the resource path should be indicated with a colon(`:`) prefix. Similarly, Rest parameters should be indicated with a double colon(`::`) prefix.
+
+***Example:***
+
+Let's consider the following client example.
+
+***main.bal***
+
+```ballerina
+public type Employee record {|
+    readonly string id;
+    string firstName;
+    string lastName;
+    string gender;
+|};
+
+EmpClient empClient = new ();
+
+public client class EmpClient {
+    map<Employee> employees;
+
+    function init() {
+        self.employees = {};
+    }
+
+    resource function get employees/[string... args]() returns string {
+        return "Welcome to the Employees API";
+    }
+
+    resource function get employee/[string id]() returns Employee? {
+        return self.employees[id];
+    }
+
+    resource function get employee/welcome/[string id](string firstName, string lastName) returns string {
+        return "Welcome " + firstName + " " + lastName + ". your ID is " + id;
+    }
+}
+```
+
+##### Stub to return a specific value with a resource method
+
+***main_test.bal***
+ 
+This test stubs the behavior of the below resource method to return a specific value in 4 ways:
+
+```ballerina
+    resource function get employee/welcome/[string id](string firstName, string lastName) returns string {
+        return "Welcome " + firstName + " " + lastName + ". your ID is " + id;
+    }
+```
+
+1. Stubbing to return a specific value in general
+2. Stubbing to return a specific value based on the path parameter
+3. Stubbing to return a specific value based on the method arguments
+4. Stubbing to return a specific value based on the path parameter and the method arguments
+
+```ballerina
+@test:Config {}
+function testWelcomeEmployee() {
+    empClient = test:mock(EmpClient);
+    // Stubbing to return a specific value in general
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").thenReturn("Stub_1");
+
+    // Stubbing to return a specific value on a specific path parameter
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").withPathParameters({id: ""}).thenReturn("Stub_2");
```

Comment:
```suggestion
    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get")
        .withPathParameters({id: ""}).thenReturn("Stub_2");
```

Reply:
In another PR, I was asked not to use line breaks as it automatically adjusts.  This is showing properly in actual website view. So, I think, we don't need this line break